index.js

```diff
@@ -48,6 +56,19 @@ var cacheControlNoTransformRegExp = /(?:^|,)\s*?no-transform\s*?(?:,|$)/
 function compression (options) {
   var opts = options || {}
 
+  if (hasBrotliSupport) {
```

Comment:
It may be a good opportunity to stop passing through `opts` blindly to each method and instead define it as `brotliParams`?

Reply:
I think in this case it makes sense to assign it this way, since the options for each compression method have not been separated yet, or maybe I misunderstood your idea.

https://nodejs.org/api/zlib.html#class-brotlioptions

Reply:
I think if we follow this suggestion it would also remove the need for the `objectAssign` addition right? because we could just manually enumerate the options for each compression method.

Reply:
If I'm understanding the idea correctly, the function would look something like this:

```javascript
function compression (options) {
  var gzipOptions = {
    level: opts.level,
    memLevel: opts.memLevel,
    strategy: opts.strategy
  }

  var brotliOptions = {
    level: opts.level,
    memLevel: opts.memLevel,
    strategy: opts.strategy
    params: opts.brotliParams
  }
//...

  stream = method === 'gzip'
        ? zlib.createGzip(gzipOptions)
        : method === 'br'
          ? zlib.createBrotliCompress(brotliOptions)
          : zlib.createDeflate(gzipOptions)

```

Reply:
Don't create the objects unless you are going to use them, but yes in general this is what I was thinking.

Reply:
@wesleytodd, enumerating the options doesn't seem like a good idea to me, as it adds unnecessary maintenance to the package. I think adding the Brotli option separately should be sufficient for now.